homeassistant/components/mochad.py

```diff
@@ -64,15 +64,26 @@ def start_mochad(event):
 class MochadCtrl:
     """Mochad controller."""
 
-    def __init__(self, host, port):
-        """Initialize a PyMochad controller."""
-        super(MochadCtrl, self).__init__()
+    def __init__(self, hass, host, port):
+        """Initialize a PyMochad send-receive controller."""
         self._host = host
         self._port = port
 
-        from pymochad import controller
+        from pymochad_mqtt.controller import PyMochadMqtt
+
+        def mqtt_pub_callback(topic, payload, qos, retain):
+            """Call MQTT publish function."""
+            mqtt.publish(hass, topic, payload, qos, retain)
 
-        self.ctrl = controller.PyMochad(server=self._host, port=self._port)
+        self.ctrl_recv = PyMochadMqtt(mochad_server=self._host,
+                                      mochad_port=self._port,
+                                      mqtt_pub_callback=mqtt_pub_callback)
+        self.ctrl_recv.start()
```

Comment:
There should be no side effects in the init method. Please start the thread outside of init.

Reply:
This doesn't look resolved.